pkg/patch/patch_test.go

```diff
@@ -291,3 +299,236 @@ func TestGetOSVersion(t *testing.T) {
+					OS: unversioned.OS{
+						Type:    "Linux",
+						Version: "5.0.1",
+					},
+					Config: unversioned.Config{
+						Arch: "x86_64",
+					},
+				},
+				Updates: []unversioned.UpdatePackage{
+					{Name: "package1"},
+					{Name: "package2"},
+					{Name: "package3"},
+				},
+			},
+		},
+		{
+			name:    "AllErrorPackages",
+			updates: updates,
+			errPkgs: []string{"package1", "package2", "package3"},
+			expected: &unversioned.UpdateManifest{
+				Metadata: unversioned.Metadata{
+					OS: unversioned.OS{
+						Type:    "Linux",
+						Version: "5.0.1",
+					},
+					Config: unversioned.Config{
+						Arch: "x86_64",
+					},
+				},
+				Updates: []unversioned.UpdatePackage{},
+			},
+		},
+		{
+			name:    "SomeErrorPackages",
+			updates: updates,
+			errPkgs: []string{"package1"},
+			expected: &unversioned.UpdateManifest{
+				Metadata: unversioned.Metadata{
+					OS: unversioned.OS{
+						Type:    "Linux",
+						Version: "5.0.1",
+					},
+					Config: unversioned.Config{
+						Arch: "x86_64",
+					},
+				},
+				Updates: []unversioned.UpdatePackage{
+					{Name: "package2"},
+					{Name: "package3"},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := updateManifest(tc.updates, tc.errPkgs)
+			if !reflect.DeepEqual(actual, tc.expected) {
+				t.Errorf("TestUpdateManifest(%v, %v): expected %v, actual %v", tc.updates, tc.errPkgs, tc.expected, actual)
+			}
+		})
+	}
+}
+
+func TestHandleError(t *testing.T) {
+	tests := []struct {
+		name    string
+		err     error
+		wantErr bool
+	}{
+		{
+			name:    "no error",
+			err:     nil,
+			wantErr: false,
+		},
+		{
+			name:    "test error",
+			err:     errors.New("test error"),
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ch := make(chan error, 1)
+			defer close(ch)
+
+			_, err := handleError(ch, tt.err)
+
+			select {
+			case chErr := <-ch:
+				if (chErr == nil && tt.wantErr) || (chErr != nil && !tt.wantErr) {
+					t.Errorf("Error channel did not return expected error, got: %v, want: %v", chErr, tt.err)
+				}
+			default:
+				if tt.wantErr {
+					t.Error("Expected handleError to send error to error channel but it did not")
+				}
+			}
+
+			if (err == nil && tt.wantErr) || (err != nil && !tt.wantErr) {
+				t.Errorf("handleError() error = %v, wantErr = %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+// define a mock reader.
+type mockReader struct {
+	data []byte
+	err  error
+}
+
+func (mr *mockReader) Read(p []byte) (int, error) {
+	copy(p, mr.data)
+	return len(mr.data), mr.err
+}
+
+func TestDockerLoad(t *testing.T) {
+	ctx := context.TODO()
+
+	testCases := []struct {
+		name      string
+		pipeR     io.Reader
+		mockCmd   *exec.Cmd
+		expectErr bool
+	}{
+		{
+			name:      "Unrecognized image format",
+			pipeR:     &mockReader{nil, errors.New("unrecognized image format")},
+			mockCmd:   exec.Command("echo", "test"),
+			expectErr: true,
```

Comment:
would it be possible to add a test case expectErr is false?

Reply:
When `dockerLoad()` is called, it takes in a TAR header as a parameter and I was unable to find a way to extract this data to create a successful test. Looking at it in the debugger, there's no clear data for me to extract and put into the tests either.

If you or Sertaç happen to know a valid TAR header or how we can access one, I'll happily add a new test case 😄